Validation/MuonME0Validation/plugins/MuonME0DigisHarvestor.cc

```diff
@@ -0,0 +1,247 @@
+
+  for (int i=1; i <= denum->GetNbinsX(); i++) {
+
+    double nNum = num->GetBinContent(i);
+    double nDenum = denum->GetBinContent(i);
+    if ( nDenum == 0 || nNum ==0  ) {
+      continue;
+    }
+    if ( nNum > nDenum ) {
+      double temp = nDenum;
+      nDenum = nNum;
+      nNum = temp;
+      edm::LogWarning("MuonME0DigisHarvestor")<<"Alert! specific bin's num is bigger than denum "<<i<<" "<<nNum<<" "<<nDenum;
+    }
+    const double effVal = nNum/nDenum;
+    efficHist->SetBinContent(i, effVal);
+    efficHist->SetBinEntries(i,1);
+    efficHist->SetBinError(i,0);
+    const double errLo = TEfficiency::ClopperPearson((int)nDenum,(int)nNum,0.683,false);
+    const double errUp = TEfficiency::ClopperPearson((int)nDenum,(int)nNum,0.683,true);
+    const double errVal = (effVal - errLo > errUp - effVal) ? effVal - errLo : errUp - effVal;
+    efficHist->SetBinError(i, sqrt(effVal * effVal + errVal * errVal));
+  }
+  return efficHist;
+}
+
+
+void MuonME0DigisHarvestor::ProcessBooking( DQMStore::IBooker& ibooker, DQMStore::IGetter& ig, std::string nameHist, TH1F* num, TH1F* den )
+{
+    
+  if( num !=nullptr && den !=nullptr ) {
+      
+    TProfile* profile = ComputeEff(num, den, nameHist);
+
+    TString x_axis_title = TString(num->GetXaxis()->GetTitle());
+    TString title  = TString::Format("Digi Efficiency;%s;Eff.",x_axis_title.Data());
+
+    profile->SetTitle( title.Data());
+    ibooker.bookProfile( profile->GetName(),profile);
+      
+    delete profile;
+      
+  }
+  else {
+      
+    edm::LogWarning("MuonME0DigisHarvestor")<<"Can not find histograms";
+    if ( num == nullptr) edm::LogWarning("MuonME0DigisHarvestor")<<"num not found";
+    if ( den == nullptr) edm::LogWarning("MuonME0DigisHarvestor")<<"den not found";
+      
+  }
+  return;
+    
+}
+
+
+TH1F* MuonME0DigisHarvestor::ComputeBKG(TH1F* hist1, TH1F* hist2, std::string nameHist)
+{
+ 
+    std::string name = "rate_"+nameHist;
+    hist1->SetName(name.c_str());
+    for (Int_t bin = 1; bin<=hist1->GetNbinsX();++bin){
+        
+        Double_t R_min = hist1->GetBinCenter(bin) - 0.5 * hist1->GetBinWidth(bin);
+        Double_t R_max = hist1->GetBinCenter(bin) + 0.5 * hist1->GetBinWidth(bin);
+        
+        Double_t Area = (TMath::Pi() * ( TMath::Power(R_max,2) - TMath::Power(R_min,2) ));
+        hist1->SetBinContent(bin,(hist1->GetBinContent(bin))/Area);
+        hist1->SetBinError(bin,(hist1->GetBinError(bin))/Area);
+        
+    }
+
+    int nEvts = hist2->GetEntries();
+    float scale = 6*2*nEvts*9*25e-9;
+    hist1->Scale(1.0/scale);
+    return hist1;
+    
+}
+
+
+void MuonME0DigisHarvestor::ProcessBookingBKG( DQMStore::IBooker& ibooker, DQMStore::IGetter& ig, std::string nameHist, TH1F* hist1, TH1F* hist2 )
+{
+    
+    if( hist1 !=nullptr && hist2 !=nullptr ) {
+
+        TH1F* rate = ComputeBKG(hist1, hist2, nameHist);
```

Comment:
Actually, it shouldn't. It's not cloned. ComputeBKG returns already existing histogram (hist1), delete here will/does cause a use-after-free later on.